app/serializers/api/tracks_serializer.rb

```diff
@@ -0,0 +1,98 @@
+class API::TracksSerializer
+  include Rails.application.routes.url_helpers
+  include Mandate
+
+  def initialize(tracks, user = nil)
+    @tracks = tracks
+    @user = user
+  end
+
+  def to_hash
+    {
+      tracks: tracks.map do |track|
+        data_for_track(track).merge(user_data_for_track(track))
+      end
+    }
+  end
+
+  private
+  attr_reader :tracks, :user
+
+  def data_for_track(track)
```

Comment:
What determines when something gets memoized? This information is also quite static.

Reply:
memoize doesn't work on methods that take params, as it memoizes the result of the method the first time it comes across it.

The result of this method is only used once in the class, so it wouldn't be appropriate to memoize it.

Reply:
Okay!